feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/signer/proxy/ProxiedSigner.kt

```diff
@@ -0,0 +1,112 @@
+package io.novafoundation.nova.feature_account_impl.data.signer.proxy
+
+import io.novafoundation.nova.common.base.errors.SigningCancelledException
+import io.novafoundation.nova.common.data.secrets.v2.SecretStoreV2
+import io.novafoundation.nova.common.utils.chainId
+import io.novafoundation.nova.common.utils.toCallInstance
+import io.novafoundation.nova.feature_account_api.data.repository.ProxyRepository
+import io.novafoundation.nova.feature_account_api.data.signer.SignerProvider
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.ProxyAccount.ProxyType
+import io.novafoundation.nova.feature_account_api.domain.model.requireAccountIdIn
+import io.novafoundation.nova.feature_account_api.presenatation.account.proxy.ProxySigningPresenter
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.signer.SignedExtrinsic
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.signer.SignedRaw
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.signer.Signer
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.signer.SignerPayloadExtrinsic
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.signer.SignerPayloadRaw
+
+class ProxiedSignerFactory(
+    private val secretStoreV2: SecretStoreV2,
+    private val chainRegistry: ChainRegistry,
+    private val accountRepository: AccountRepository,
+    private val proxySigningPresenter: ProxySigningPresenter,
+    private val proxyRepository: ProxyRepository
+) {
+
+    fun create(metaAccount: MetaAccount, signerProvider: SignerProvider): ProxiedSigner {
+        return ProxiedSigner(
+            metaAccount,
+            chainRegistry,
+            accountRepository,
+            signerProvider,
+            proxySigningPresenter,
+            proxyRepository
+        )
+    }
+}
+
+class ProxiedSigner(
+    private val proxiedMetaAccount: MetaAccount,
+    private val chainRegistry: ChainRegistry,
+    private val accountRepository: AccountRepository,
+    private val signerProvider: SignerProvider,
+    private val proxySigningPresenter: ProxySigningPresenter,
+    private val proxyRepository: ProxyRepository
+) : Signer {
+
+    override suspend fun signExtrinsic(payloadExtrinsic: SignerPayloadExtrinsic): SignedExtrinsic {
+        val proxyMetaAccount = getProxyMetaAccount()
+
+        requestResume(proxyMetaAccount)
```

Comment:
I found `requestResume` to be a little bit unclear, maybe we can rename it to `acknowledgeProxyOperation`?